battle-engine.js

```diff
@@ -1052,11 +1052,9 @@ BattlePokemon = (function () {
 		if (!sourceEffect && this.battle.effect) sourceEffect = this.battle.effect;
 		if (!source && this.battle.event && this.battle.event.target) source = this.battle.event.target;
 		item = this.getItem();
-		if (this.battle.runEvent('UseItem', this, null, null, item) && this.battle.runEvent('EatItem', this, null, null, item)) {
```

Comment:
`EatItem` is actually a very important for making sure berries don't get consumed out of order, and is used in quite a few cases:

- Unnerve blocking all berries
- Heal Block blocking healing berries

The way the order is now, an eating message will appear, the effect will trigger, and then Unnerve will block the item from being "consumed" and the item will stay as held. Berry Pouch probably needs to activate in an event that occurs after `EatItem`, perhaps in `AfterUseItem`? We'll still have to do some extra checks for Bug Bite and whatnot (because those are not fixed yet)